py-polars/src/dataframe.rs

```diff
@@ -1356,17 +1357,21 @@ impl PyDataFrame {
         Ok(hash.into_series().into())
     }
 
-    pub fn transpose(&self, include_header: bool, names: &str) -> PyResult<Self> {
-        let mut df = self.df.transpose().map_err(PyPolarsErr::from)?;
-        if include_header {
-            let s = Utf8Chunked::from_iter_values(
-                names,
-                self.df.get_columns().iter().map(|s| s.name()),
-            )
-            .into_series();
-            df.insert_at_idx(0, s).unwrap();
-        }
-        Ok(df.into())
+    #[pyo3(signature = (keep_names_as, column_names))]
+    pub fn transpose(&self, keep_names_as: Option<&str>, column_names: &PyAny) -> PyResult<Self> {
+        // It doesn't automatically translate Python types to Either :-(
```

Comment:
```suggestion
```

Reply:
This is expected as `Either` is a third party crate pyo3 will not have any conversion to their types and so don't we because of the orphan rule.

Reply:
I actually didn't realize that. Fun!